File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        size = len(nums)
        dp = [0]*size
        if size <= 2:
            return max(nums)
        dp[size-1] = nums[size-1]
        dp[size-2] = max(nums[size-2],dp[size-1])
        for i in range(size-3 , -1, -1):
            dp[i] = max(dp[i+1], nums[i]+dp[i+2])

        return dp[0]


        

File: Data Structures & Algorithms/three-integer-sum/submission-2.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Set<List<Integer>> resultSet = new HashSet<List<Integer>>();
        List<List<Integer>> resultList = new ArrayList<List<Integer>>();

        Arrays.sort(nums);
        for(int i=0;i<nums.length-2;i++){
            int left = i+1;
            int right = nums.length-1;
            while(left < right){
                List<Integer> output = new ArrayList<>();
                int finalsum = nums[i] + nums[left] + nums[right];
                if(finalsum == 0){
                    output.add(nums[i]);
                    output.add(nums[left]);
                    output.add(nums[right]);
                    resultSet.add(output);
                   // output.clear();
                    System.out.println(nums[i]+" "+nums[left]+" "+nums[right]+" "+finalsum);
                    left++;
                    right--;
                }else if(finalsum > 0){
                    right--;
                }else{
                    left++;
                }
            }
        }
        resultList.addAll(resultSet);
        return resultList;
    }
}
